src/serialbox/core/Compiler.h

```diff
@@ -66,7 +66,7 @@
 ///
 /// On compilers which support it, expands to an expression which states that it is undefined
 /// behaviour for the compiler to reach this point. Otherwise is not defined.
-#if __has_builtin(__builtin_unreachable) || SERIALBOX_GNUC_PREREQ(4, 5, 0)
+#if (__has_builtin(__builtin_unreachable) || SERIALBOX_GNUC_PREREQ(4, 5, 0) ) && ! defined(_CRAYC)

```

Comment:
The Cray compiler seems to enter this branch, but does not provide a symbol for `__builtin_unreachable()` so that we get an undefined reference at link time.
This chunk  bypasses this branch for the Cray compiler